### ISSUE TITLE
GetPropertyMethod not called if set to a custom name

### ISSUE THREAD

**Issue by mmuffins** [tmds/Tmds.DBus#298]
Hi,
I'm trying to implement a simple mediaplayer service and can't figure out how to properly set custom method names for the `DBusInterface` attribute.

Here's a simple example where I create a interface inheriting from `IDBusObject` for `org.mpris.MediaPlayer2`, and specify `GetPropertyMethod` to point to `MplayerGetAsync`:

```csharp
using System;
using System.Collections.Generic;
using System.Runtime.CompilerServices;
using System.Threading;
using System.Threading.Tasks;
using Tmds.DBus;

[assembly: InternalsVisibleTo(Tmds.DBus.Connection.DynamicAssemblyName)]
namespace MyMediaPlayer
{
    [Dictionary]
    public class MediaPlayerProperties
    {
        public readonly bool CanQuit = false;
        public readonly bool CanRaise = false;
        public readonly bool HasTrackList = false;
        public string Identity = "test";
        public string DesktopEntry = "test";
        public string[] SupportedUriSchemes = Array.Empty<string>();
        public string[] SupportedMimeTypes = Array.Empty<string>();
    }


    [DBusInterface("org.mpris.MediaPlayer2",
        GetPropertyMethod = "MplayerGetAsync",
        PropertyType = typeof(MediaPlayerProperties)
    )]
    interface IMediaPlayer2 : IDBusObject
    {
        public Task<MediaPlayerProperties> GetAllAsync();
        public Task<object> MplayerGetAsync(string prop);
        public Task SetAsync(string property, object value);

        Task RaiseAsync();
        Task QuitAsync();

    }

    public class MediaPlayer : IMediaPlayer2
    {
        readonly static MediaPlayerProperties mediaPlayerProperties = new();

        public event Action<PropertyChanges> OnPropertiesChanged;

        public ObjectPath ObjectPath => new ObjectPath("/org/mpris/MediaPlayer2");

        public MediaPlayer() { }

        public Task<MediaPlayerProperties> GetAllAsync()
        {
            Console.WriteLine("GetAllAsync");
            return Task.FromResult(mediaPlayerProperties);
        }

        public Task<object> MplayerGetAsync(string prop)
        {
            Console.WriteLine("GetAsync");
            return Task.FromResult<object>(false);
        }

        public Task SetAsync(string property, object value)
        {
            Console.WriteLine("SetAsync.");
            OnPropertiesChanged?.Invoke(PropertyChanges.ForProperty(property, value));
            return Task.CompletedTask;
        }

        public Task RaiseAsync()
        {
            Console.WriteLine("RaiseAsync");
            return Task.CompletedTask;
        }

        public Task QuitAsync()
        {
            Console.WriteLine("QuitAsync");
            return Task.CompletedTask;
        }

        public Task<IDisposable> WatchPropertiesAsync(Action<PropertyChanges> handler)
        {
            return SignalWatcher.AddAsync(this, nameof(OnPropertiesChanged), handler);
        }
    }
}

namespace MyMediaPlayer
{
    class Program
    {
        private static async Task Main(string[] args)
        {
            using var connection = new Connection(Address.Session);
            await connection.ConnectAsync();

            var mediaPlayer = new MediaPlayer();
            await connection.RegisterObjectAsync(mediaPlayer);
            await connection.RegisterServiceAsync("org.mpris.MediaPlayer2.myplayer");

            Console.WriteLine("Media player service registered.");
            
            await Task.Delay(-1); // Keep the application running
        }
    }
}
```

If I run this code and then try to read any of the properties d-feet nothing happens. No response, no log entry, no exception. The `MplayerGetAsync` method is never called.

If I now change `GetPropertyMethod` in the DBusInterface attribute to the default `GetAsync` name (or omit it), and rename all methods accordingly, everything works perfectly and I can see in the log that the `GetAsync` method in the main `MediaPlayer` class is called.

I also tried this with the `GetAllPropertiesMethod` property with the same result, the method is only called if I leave it at its default value.

What's the correct way to provide a custom name for those methods?

**Comment by tmds** [comment]
I can't reproduce this issue.

I tried to reproduce the issue by setting `GetPropertyMethod` in the test suite (diff below), and all tests passed.

In the Tmds.DBus code, you'll find the `"GetAsync"` literal only to initialize `GetPropertyMethod`. So if `GetPropertyMethod` is changed, the library should match with the new name.

What version of .NET are you using?

```diff
diff --git a/test/Tmds.DBus.Tests/ConnectionTests.cs b/test/Tmds.DBus.Tests/ConnectionTests.cs
index 3731d2d..cd11ecc 100644
--- a/test/Tmds.DBus.Tests/ConnectionTests.cs
+++ b/test/Tmds.DBus.Tests/ConnectionTests.cs
@@ -79,14 +79,14 @@ public async Task Properties()
             var properties = await proxy.GetAllAsync();
             Assert.Equal(dictionary, properties);
 
-            var val1 = await proxy.GetAsync("key1");
+            var val1 = await proxy.GetFooAsync("key1");
             Assert.Equal(1, val1);
 
             var tcs = new TaskCompletionSource<PropertyChanges>();
             await proxy.WatchPropertiesAsync(_ => tcs.SetResult(_));
             await proxy.SetAsync("key1", "changed");
 
-            var val1Changed = await proxy.GetAsync("key1");
+            var val1Changed = await proxy.GetFooAsync("key1");
             Assert.Equal("changed", val1Changed);
 
             var changes = await tcs.Task;
diff --git a/test/Tmds.DBus.Tests/DaemonTests.cs b/test/Tmds.DBus.Tests/DaemonTests.cs
index e0d0064..c43aea2 100644
--- a/test/Tmds.DBus.Tests/DaemonTests.cs
+++ b/test/Tmds.DBus.Tests/DaemonTests.cs
@@ -89,14 +89,14 @@ public async Task Properties()
                 var properties = await proxy.GetAllAsync();
 
                 Assert.Equal(dictionary, properties);
-                var val1 = await proxy.GetAsync("key1");
+                var val1 = await proxy.GetFooAsync("key1");
                 Assert.Equal(1, val1);
 
                 var tcs = new TaskCompletionSource<PropertyChanges>();
                 await proxy.WatchPropertiesAsync(_ => tcs.SetResult(_));
                 await proxy.SetAsync("key1", "changed");
 
-                var val1Changed = await proxy.GetAsync("key1");
+                var val1Changed = await proxy.GetFooAsync("key1");
                 Assert.Equal("changed", val1Changed);
 
                 var changes = await tcs.Task;
diff --git a/test/Tmds.DBus.Tests/IPropertyObject.cs b/test/Tmds.DBus.Tests/IPropertyObject.cs
index 7c2b65d..0368c63 100644
--- a/test/Tmds.DBus.Tests/IPropertyObject.cs
+++ b/test/Tmds.DBus.Tests/IPropertyObject.cs
@@ -5,11 +5,12 @@
 
 namespace Tmds.DBus.Tests
 {
-    [DBusInterface("tmds.dbus.tests.PropertyObject")]
+    [DBusInterface("tmds.dbus.tests.PropertyObject",
+        GetPropertyMethod = "GetFooAsync")]
     public interface IPropertyObject : IDBusObject
     {
         Task<IDictionary<string, object>> GetAllAsync();
-        Task<object> GetAsync(string prop);
+        Task<object> GetFooAsync(string prop);
         Task SetAsync(string prop, object val);
         Task<IDisposable> WatchPropertiesAsync(Action<PropertyChanges> handler);
     }
diff --git a/test/Tmds.DBus.Tests/PropertyObject.cs b/test/Tmds.DBus.Tests/PropertyObject.cs
index 7cedd76..62a46b6 100644
--- a/test/Tmds.DBus.Tests/PropertyObject.cs
+++ b/test/Tmds.DBus.Tests/PropertyObject.cs
@@ -28,7 +28,7 @@ public Task<IDictionary<string, object>> GetAllAsync()
             return Task.FromResult(_properties);
         }
 
-        public Task<object> GetAsync(string prop)
+        public Task<object> GetFooAsync(string prop)
         {
             return Task.FromResult(_properties[prop]);
         }

```

**Comment by mmuffins** [comment]
Sorry, should have mentioned that. I'm using the tmds.dbus and tmds.dbus.protocol packages with version 0.19.0 on a .net 8 application in debian trixie/sid with gnome 46.

One thing to note is that the application itself has no issue accessing the `MplayerGetAsync` method. If I add `await mediaPlayer.MplayerGetAsync("test");` in the main method, it is called without any issues. 

But if I let my application run, then open org.mpris.MediaPlayer2.myplayer in d-feet, and try to read a property of the org.mpris.MediaPlayer2 interface by double clicking it, that's when it's not working for me. Running dbus-monitor while doing that shows a dbus error saying that the `get` method was not found:

```
method call time=1722781160.608194 sender=:1.320 -> destination=:1.590 serial=1160 path=/org/mpris/MediaPlayer2; interface=org.freedesktop.DBus.Properties; member=Get
   string "org.mpris.MediaPlayer2"
   string "CanQuit"
error time=1722781160.608465 sender=:1.590 -> destination=:1.320 error_name=org.freedesktop.DBus.Error.UnknownMethod reply_serial=1160
   string "Method "Get" with signature "ss" on interface "org.freedesktop.DBus.Properties" doesn't exist"
```

Changing back the method to the default `GetAsync` name and repeating the test in d-feet changes the output of dbus-monitor to this:
```
method call time=1722781442.669217 sender=:1.320 -> destination=:1.634 serial=1233 path=/org/mpris/MediaPlayer2; interface=org.freedesktop.DBus.Properties; member=Get
   string "org.mpris.MediaPlayer2"
   string "CanQuit"
method return time=1722781442.669518 sender=:1.634 -> destination=:1.320 serial=26 reply_serial=1233
   variant       string "false"
```

**Comment by tmds** [comment]
The fix is part of the 0.20.0 release, which was just uploaded to nuget.org.
